internal/api/genapi/input.go

```diff
@@ -520,6 +520,17 @@ var inputStructs = []*structInfo{
 			mapstructureConversionTemplate,
 		},
 	},
+	{
+		inProto:     &credentials.KvAttributes{},
+		outFile:     "credentials/kv_attributes.gen.go",
+		subtypeName: "KvCredential",
+		fieldOverrides: []fieldInfo{
+			{
+				Name:        "Secrets",
+				SkipDefault: true,
+			},
+		},
```

Comment:
I would need to dig into the usage within the gen path but this does not include some of the fields in other credentials:
```
parentTypeName: "Credential",
templates: []*template.Template{
	mapstructureConversionTemplate,
},
```

Reply:
good catch, I will take a look at this & check if the api tests still pass

Reply:
I added this change and the tests are still passing. I will added more checks to validate the generated functions.